src/pages/home/report/ReportActionItem.js

```diff
@@ -117,6 +120,21 @@ function ReportActionItem(props) {
         focusTextInputAfterAnimation(textInputRef.current, 100);
     }, [isDraftEmpty]);
 
+    // Hide the message if it is being moderated for a higher offense, or is hidden by a moderator
+    // Removed messages should not be shown anyway and should not need this flow
+    useEffect(() => {
+        if (_.isEmpty(props.action.moderationDecisions)) {
+            return;
+        }
+
+        // Right now we are only sending the latest moderationDecision to the frontend even though it is an array
+        const latestDecision = props.action.moderationDecisions[0];
+        if (latestDecision.decision === CONST.MODERATION.MODERATOR_DECISION_PENDING_HIDE || latestDecision.decision === CONST.MODERATION.MODERATOR_DECISION_HIDDEN) {
+            setIsHidden(true);
+        }
+        setModerationDecision(latestDecision.decision);
+    }, [props.action, moderationDecision]);
```

Comment:
Can we check props.action.message so this doesn't re-render as much? Also it doesn't seem like we need to add a case for moderationDecision. It only changes when the prop.action.message changes, and then it would cause another run of this since the moderationDecision changed

Reply:
yeah i agree on moderationDecision... but the linter did not haha. I'll see if switching over to the right format for the message object fixes that.